### PR TITLE
unwind: make `build.rs` do clean rebuilds

### PR DESCRIPTION
**Pull request by gburgessiv**
If the build by this was interrupted or otherwise left incomplete, there
could be object files hanging around, which triggers an assertion later.
Since we unconditionally rebuild everything here anyway, start with a
clean slate each build.

r? @Mark-Simulacrum

### CODE REVIEW

File: library/unwind/build.rs [hunk 1]
```diff
@@ -76,6 +88,16 @@ mod llvm_libunwind {
         let mut cpp_cfg = cc::Build::new();
         let root = Path::new("../../src/llvm-project/libunwind");
 
+        // We depend on starting with a fresh build directory each time.
```
Comment (Mark-Simulacrum) [inline library/unwind/build.rs:91]: Hm, I'm wondering if we can get away with just deleting the directory as a whole here.

Comment (gburgessiv) [reply library/unwind/build.rs:80]: Works on my machine; PTAL :)

**Review by gburgessiv — COMMENTED** [review]
Thanks for the feedback!

**Comment by gburgessiv** [comment]
TIL my `cargo fmt` config doesn't mesh with this repo's. Ran `./x.py fmt` to try to fix the things -- `./x.py test` no longer complains about formatting issues :)

**Comment by gburgessiv** [comment]
Looks like the changes that caused the merge conflict also added a check to tell objects generated from C++ sources apart from objects generated from C sources. Hence, this should be fixed already, so I'll close it out. Thanks again for the review! :)

**Pull request closed without merging**